power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/newton_raphson_se_solver.hpp

```diff
@@ -170,31 +178,312 @@ template <bool sym> class NewtonRaphsonSESolver {
+                        block_j += jacobian_diagonal_component(jac_complex_unit_other_tt, calculated_power);
+
+                        if (type == YBusElementType::btt) {
+                            multiply_add_jacobian_blocks(block, rhs_block, block_j, block_j, measured_power,
+                                                         calculated_power);
+
+                        } else if (type == YBusElementType::btf) {
+                            auto const jac_complex_unit_other_tf = dot(jac_complex_tf, abs_uj_inv);
+                            auto const block_i = calculate_jacobian(jac_complex_tf, jac_complex_unit_other_tf);
+                            multiply_add_jacobian_blocks(block, rhs_block, block_j, block_i, measured_power,
+                                                         calculated_power);
+                        }
+                    }
+                }
+
+                // fill block with injection measurement
+                // injection measurement exist
+                if (measured_value.has_bus_injection(row)) {
+                    auto const& yij = y_bus.admittance()[data_idx];
+                    auto const jac_complex_ft = jac_complex_intermediate_form(yij, ui_uj_conj);
+                    auto const jac_complex_unit_other_ft = dot(jac_complex_ft, abs_uj_inv);
+                    auto const jac_complex_unit_self_ft = dot(jac_complex_ft, abs_ui_inv);
+                    auto const calculated_power = sum_row(jac_complex_ft);
+
+                    auto const injection_jac = calculate_jacobian(jac_complex_ft, jac_complex_unit_other_ft);
+                    add_injection_jacobian(block, injection_jac);
+
+                    // add summed component to the diagonal block
+                    auto const injection_jac_diagonal =
+                        jacobian_diagonal_component(jac_complex_unit_self_ft, calculated_power);
+                    add_injection_jacobian(diag_block, injection_jac_diagonal);
+                    // subtract f(x) incrementally
+                    rhs_block.tau_p() -= real(calculated_power);
+                    rhs_block.tau_q() -= imag(calculated_power);
+
+                    // R_ii = -variance, only diagonal
+                    if (row == col) {
+                        // assign variance to diagonal of 3x3 tensor, for asym
+                        auto const& injection = measured_value.bus_injection(row);
+                        rhs_block.tau_p() += injection.value.real();
+                        rhs_block.tau_q() += injection.value.imag();
+
+                        block.r_P_theta() -=
+                            RealTensor<sym>{RealValue<sym>{RealValue<sym>{1.0} / injection.p_variance}};
+                        block.r_Q_v() -= RealTensor<sym>{RealValue<sym>{RealValue<sym>{1.0} / injection.q_variance}};
+                    }
+
+                }
+                // injection measurement not exist
+                else {
```

Comment:
the if statement already suggests this and the comment already explains what should happen when there's no injection measurement

```suggestion
                } else {
```